README.md

```diff
@@ -4,104 +4,168 @@
 [paper](https://people.eecs.berkeley.edu/~apanda/assets/papers/osdi16.pdf) for information
 about the architecture and design. Currently NetBricks requires a relatively modern Linux version.
 
-Building
---------
-NetBricks can be built either using a Rust nightly build or using Rust built from the current Git head. In the later
-case we also build [`musl`](https://www.musl-libc.org/) and statically link to things. Below we provide basic instructions for both.
+Up and Running
+----------------
 
-Finally, in addition to the above options, NetBricks can also be built within a Docker container. In this case, you do
-not need to install any of the dependencies, and the final product can be run the same. However to run NetBricks you
-still need to be on a machine that is correctly configured to run DPDK (see
-[here](http://dpdk.org/doc/guides-16.07/linux_gsg/quick_start.html) for more details), and you still need to install
-Rust nightly (for libraries). Please see the [container build instructions](#container-build) for more information.
+NetBricks can built within a Docker container. In this case, you do not need to
+install any of the dependencies, and the final product can be run the same.
+However to run NetBricks you still need to be on a machine or VM that is
+correctly configured to run [DPDK](https://www.dpdk.org/) (version 17.08.1).
 
-Using Rust Nightly
-------------------
-First obtain Rust nightly. I use [rustup](https://rustup.rs), in which case the following works
+If you use our vagrant-based Developer environment, all DPDK configuration is
+done for you. We also include the [MoonGen](//github.com/williamofockham/MoonGen) traffic generator and the
+[Containernet](//github.com/containernet/containernet) virtual network simulator for testing and development.
 
+## Creating a Developer environment with `vagrant`
+
+1. Clone our [utils](//github.com/williamofockham/utils) and [moonGen](//github.com/williamofockham/MoonGen)
+   repositories into the same parent directory.
+   ```shell
+   host$ for repo in utils moonGen; do \
+           git clone --recurse-submodules git@github.com:williamofockham/${repo}.git; \
+         done
+   ```
+
+2. [Install Vagrant](https://www.vagrantup.com/docs/installation/) and
+   [VirtualBox](https://www.virtualbox.org/wiki/Downloads).
+
+3. Install the `vagrant-disksize` (required) and `vagrant-vbguest` (recommended)
+   `vagrant-reload` (required) plugins:
+   ```shell
+   host$ vagrant plugin install vagrant-disksize vagrant-vbguest vagrant-reload
+   ```
+
+4. Symlink the Vagrantfile into the parent directory.
+   ```shell
+   host$ ln -s utils/Vagrantfile
+   ```
+
+4. Boot the VM:
+   ```shell
+   host$ vagrant up
```

Comment:
following these directions, I encountered:

```
There are errors in the configuration of this machine. Please fix
the following errors and try again:

shell provisioner:
* `path` for shell provisioner does not exist on the host system: /Users/pclear200/dev/github/williamofockham/vm-kernel-upgrade.sh
* `path` for shell provisioner does not exist on the host system: /Users/pclear200/dev/github/williamofockham/vm-setup.sh
```

Reply:
Also, the name collides with our internal dev environment...

```
A VirtualBox machine with the name 'ubuntu-xenial-williamofockham' already exists.
Please use another name or delete the machine with the existing
name, and try again.
```

Should we use a different name for this virtual box vs. internal?

Reply:
Agreed on the name @pauljamescleary. The error you saw is due to needing the PRs in https://github.com/williamofockham/utils/pull/37, https://github.com/williamofockham/libmoon/pull/6, and https://github.com/williamofockham/MoonGen/pull/23.